alg/gdaltransformer.cpp

```diff
@@ -615,62 +633,60 @@ GDALSuggestedWarpOutput2( GDALDatasetH hSrcDS,
                     break;
                 }
 
-                double dfRatio = (i % (nSteps + 1)) * dfStep;
+                dfRatio = (i % nStepsPlusOne) * dfStep;
                 if( dfRatio > 0.99 )
                     dfRatio = 1.0;
 
-                double dfExpectedX = 0.0;
-                double dfExpectedY = 0.0;
-                if( i < nSteps + 1 )
-                {
+                dfExpectedX = 0.0;
+                dfExpectedY = 0.0;
+                switch (i / nStepsPlusOne) {
+                case 1:
+                    dfExpectedY = nInYSize;
+                    //avoid -Wimplicit-fallthrough
                     dfExpectedX = dfRatio * nInXSize;
-                }
-                else if( i < 2 * (nSteps + 1) )
-                {
+                    break;
+                case 0:
                     dfExpectedX = dfRatio * nInXSize;
-                    dfExpectedY = nInYSize;
-                }
-                else if( i < 3 * (nSteps + 1) )
-                {
-                    dfExpectedY   = dfRatio * nInYSize;
-                }
-                else
-                {
-                    dfExpectedX   = nInXSize;
-                    dfExpectedY   = dfRatio * nInYSize;
+                    break;
+                case 2:
+                    dfExpectedY = dfRatio * nInYSize;
+                    break;
+                default:
+                    dfExpectedX = nInXSize;
+                    dfExpectedY = dfRatio * nInYSize;
```

Comment:
I'm a bit skeptical about using switch() here.
At the very least, the order of the case should be reorderd into the logical order 0, 1, 2, default. And then you can remove the // avoid -Wimplicit-fallthrough comment

Reply:
OK, `if else` is more human readable than `switch`, I'll revert this change